### PR TITLE
Improvements to Conda build

### PR DESCRIPTION
**Pull request by dhruvbatra**
Assorted improvements to README, build functions to keep track of version numbers, etc. 

CC: @mpiseno 

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: conda-build/linux_matrix_builder.py [hunk 1]
```diff
@@ -4,6 +4,12 @@
 import os.path as osp
 import itertools
 import os
+import time
+
+import sys
```
Comment (erikwijmans) [inline conda-build/linux_matrix_builder.py:9]: I think these also need an `import builtins`

File: conda-build/linux_matrix_builder.py [hunk 2]
```diff
@@ -38,6 +44,7 @@ def main():
         py_vers, bullet_modes, headless_modes, cuda_vers
     ):
         env = os.environ.copy()
+        env["VERSION"] = habitat_sim.__version__ + time.strftime(".%Y.%m.%d") # including a timestamp in anticipation of nightly builds
```
Comment (Skylion007) [inline conda-build/linux_matrix_builder.py:47]: Why not just do a has of the current commit? No reason to add builds that are identical.

File: conda-build/macos_matrix_builder.py [hunk 3]
```diff
@@ -34,6 +40,7 @@ def main():
 
     for py_ver, use_bullet in itertools.product(py_vers, bullet_modes):
         env = os.environ.copy()
+        env["VERSION"] = habitat_sim.__version__ + time.strftime(".%Y.%m.%d") # adding time-stamp in anticipation of nightly builds
```
Comment (Skylion007) [inline conda-build/macos_matrix_builder.py:43]: Likewise for above

File: conda-build/README.md [hunk 4]
```diff
@@ -11,11 +11,17 @@ To then download the package, run ```conda install -c aihabitat -c conda-forge h
 
 The process is almost the same for linux; there is a corresponding python script for starting things off, however we use a docker container to do the builds. There is a dockerfile in this directory that you can use to create a container.
 
-```docker build -t myContainer -f conda-build/Dockerfile .```
+```docker build -t mycontainer -f Dockerfile .```
```
Comment (Skylion007) [inline conda-build/README.md:14]: We should probably actually give it an intelligent name like: habitat_sim_conda_build

File: conda-build/linux_matrix_builder.py (hunk 2 above)
Comment (Skylion007) [reply conda-build/linux_matrix_builder.py:47]: Requires gitpython library.
```python
import git
repo = git.Repo(search_parent_directories=True)
sha = repo.head.object.hexsha
```

**Comment by dhruvbatra** [comment]
@erikwijmans @Skylion007 -- addressed. Let me know if this is ready to merge.

**Comment by Skylion007** [comment]
Sorry to go back on my previous comment, but upon reading more of the conda
doc, it suggests that the commit hash and such should be stored in the
build string, not the version number:
https://docs.conda.io/projects/conda-build/en/latest/resources/package-spec.html

I am not sure what the actual best way to version nightly releases in conda
is though.

**Comment by Skylion007** [comment]
This is how torch does it (with the devprefix):
https://github.com/pytorch/pytorch/issues/33103#issuecomment-583805093

**Comment by Skylion007** [comment]
Looks like we may have to increment the BUILD_NUMBER variable to sort the
versions properly.

**Comment by dhruvbatra** [comment]
@Skylion007 -- I added date back to version and git SHA to build string. This is similar but not identical to PyTorch. I don't think we need to overoptimize this. It's a nightly build. We don't plan on having multiple nightly builds with the same date. This seems good enough to me.

**Pull request merged**